### PR TITLE
LPS-67115 Migrating ADD_ENTRY permissions for Announcements and alerts

### PR DESCRIPTION
**Pull request by stian-sigvartsen**
Originally sent to @topolik. Replaces topolik/liferay-portal/pull/362

> This pull changes the ADD_ENTRY permission for Announcements and Alerts. Before this pull, ADD_ENTRY was a permission that was granted on a portlet basis. This means that some users could have permission to add announcements in a portlet in a page but didn't have permission to add announcements in a different page, even though both pages belonged to the same site.
> 
> With this pull we are changing the scope of the ADD_ENTRY permission so this permission will be granted on a site based and not on a portlet based. So if the user has permission to ADD_ENTRY he will have permission in every portlet and every page of the site.
> 
> We are checking the ResourcePermission table to see who was granted ADD_ENTRY permission in a portlet, and as long as the user has permisison in just one portlet in the site, we will grant them ADD_ENTRY permission at the site level.
> 
> As you are more experienced with the Liferay permissioning system we'd like you to review the upgrade that we wrote to see if there's anything that we might have missed or any use case that we didn't consider.
> 
> Let us know if you have any question.

/cc @sergiogonzalez

### CODE REVIEW

**Comment by stian-sigvartsen** [comment]
@sergiogonzalez I've created this new pull so you can track...

**Comment by sergiogonzalez** [comment]
thanks @stian-sigvartsen 👍

**Comment by stian-sigvartsen** [comment]
See suggested changes: sergiogonzalez/liferay-portal/pull/3356

File: portal-impl/src/com/liferay/portlet/announcements/service/permission/AnnouncementsEntryPermission.java [hunk 1]
```diff
@@ -103,6 +104,11 @@ public static boolean contains(
 			layout = virtualLayout.getSourceLayout();
 		}
 
+		if (actionId.equals(ActionKeys.ADD_ENTRY)) {
+			return AnnouncementsPermission.contains(
+				permissionChecker, layout.getGroupId(), actionId);
+		}
+
```
Comment (stian-sigvartsen) [inline portal-impl/src/com/liferay/portlet/announcements/service/permission/AnnouncementsEntryPermission.java]: Do we need this fallback? There are no calls to this method with ActionKeys.ADD_ENTRY. Is this to support legacy 3rd party code?

**Comment by stian-sigvartsen** [comment]
See stian-sigvartsen/liferay-portal/pull/42

**Pull request closed without merging**